daemon/execdriver/native/create.go

```diff
@@ -91,6 +91,8 @@ func generateIfaceName() (string, error) {
 func (d *driver) createNetwork(container *configs.Config, c *execdriver.Command) error {
 	if c.Network.HostNetworking {
 		container.Namespaces.Remove(configs.NEWNET)
+		container.Namespaces.Remove(configs.NEWUTS)
```

Comment:
There is validation in libcontainer about missing `NEWUTS` and non-empty hostname. So I think it is better to remove emptying `Hostname`. Also docker itself forbids `--net=host` with `--hostname`.